### PR TITLE
fix: [ANDROAPP-2786] Program rules are not filtered by program stage

### PR DESCRIPTION
**Pull request by Balcan**
[issue](https://jira.dhis2.org/browse/ANDROAPP-2786)

### CODE REVIEW

File: app/src/main/java/org/dhis2/data/forms/RulesRepository.kt [hunk 1]
```diff
@@ -22,6 +19,9 @@ import org.hisp.dhis.rules.models.RuleEnrollment
 import org.hisp.dhis.rules.models.RuleEvent
 import org.hisp.dhis.rules.models.RuleVariable
 import timber.log.Timber
+import java.util.Calendar
+import java.util.Date
+import java.util.Objects
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/data/forms/RulesRepository.kt]: Not in lexicographical order, does not comply with Kotlin code style

Comment (Balcan) [reply app/src/main/java/org/dhis2/data/forms/RulesRepository.kt]: ok...:(

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureRepositoryImpl.java [hunk 2]
```diff
@@ -139,71 +135,6 @@ public EventCaptureRepositoryImpl(Context context, FormRepository formRepository
         sectionFields = new ArrayList<>();
     }
 
-    private void loadDataElementRules(Event event) {
-        List<ProgramRule> rules = d2.programModule().programRules().byProgramUid().eq(event.program()).withProgramRuleActions().blockingGet();
-        List<ProgramRule> mandatoryRules = new ArrayList<>();
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureRepositoryImpl.java]: not needed?¿??

Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureRepositoryImpl.java]: nope

**Pull request merged**